### ISSUE TITLE
Unable to fetch IndexedSlices

### ISSUE THREAD

**Issue by kentonl** [tensorflow/tensorflow#518]
It appears that IndexedSlices cannot be fetched, as demonstrated by the following program, which is trying to inspect the gradients of an embedding matrix:

```
import tensorflow as tf

x = tf.placeholder(tf.int32)
y = tf.placeholder(tf.float32)
embeddings = tf.get_variable("embeddings", [1, 1])
loss = tf.reduce_mean(tf.square(y - tf.nn.embedding_lookup(embeddings, x)))
grads = tf.gradients(loss, embeddings)

with tf.Session() as session:
    tf.initialize_all_variables().run()
    print("Gradient value: {}".format(session.run(grads, { x : 0, y : 0.0 })))
```

Running the code above gives this error:
```
TypeError: Fetch argument <tensorflow.python.framework.ops.IndexedSlices object at 0x1053d69d0> of <tensorflow.python.framework.ops.IndexedSlices object at 0x1053d69d0> has invalid type <class 'tensorflow.python.framework.ops.IndexedSlices'>, must be a string or Tensor. (Can not convert a IndexedSlices into a Tensor or Operation.)
```

Is this result intended? If so, would there be a better way to inspect the gradients?

Thanks!

**Comment by looveu** [comment]
I got similar problems

**Comment by mrry** [comment]
One possibility is to map `tf.convert_to_tensor` over `grads` so that every element is converted to a (dense) `Tensor`. Does that work for your purposes?

**Comment by kentonl** [comment]
Yes, that would work, but it would great it can be done without losing sparsity.

**Comment by mrry** [comment]
It would certainly be possible to handle this by handling `IndexedSlices` in a similar manner to `SparseTensor` in `Session.run()`, using the subfeed/subfetch mechanism. We'd be happy to take contributions on this!

**Comment by kentonl** [comment]
Fixed by #541.
